### PR TITLE
CI for deployment to Dokku

### PR DESCRIPTION
**Pull request by esirK**
## Description
Automates image building and deployment to dokku.

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -121,5 +121,9 @@
     "*.yml": [
       "yarn format-staged"
     ]
+  },
+  "engines": {
+    "node": ">=16.0.0 <17.0.0",
+    "npm": ">=8.0.0 <9.0.0"
```
Comment (kilemensi) [inline package.json:127]: Do we need this? I think we use `yarn` mostly, no?

File: .github/workflows/build_deploy.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+      - name: Cache Docker layers
+        if: steps.version-check.outputs.changed == 'true'
+        uses: actions/cache@v3
+        with:
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          path: /tmp/.buildx-cache
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+      - name: Login to DockerHub
```
Comment (kilemensi) [inline .github/workflows/build_deploy.yml:53]:
```suggestion

      - name: Login to DockerHub
```

File: .github/workflows/build_deploy.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+          context: .
+          push: true
+          tags: "${{ env.IMAGE_NAME }}:${{ steps.version-check.outputs.version }}"
+
+      - name: Move cache
+        if: steps.version-check.outputs.changed == 'true'
+        run: |
+          rm -rf /tmp/.buildx-cache
+          mv /tmp/.buildx-cache-new /tmp/.buildx-cache
+      - name: Push to Dokku
```
Comment (kilemensi) [inline .github/workflows/build_deploy.yml:79]:
```suggestion

      - name: Push to Dokku
```

**Review by kilemensi — APPROVED** [review]
🚀 

---

My Internet seem to be too slow to build locally (2nd day today) but it looks fine. Lets just watch out for these `ARG`s [here](https://github.com/CodeForAfrica/PesaYetu/blob/1049ed97fe56885cfcc0713fd9f01fb126d7705d/Dockerfile#L10-L24). They might be needed on the pages that are completely built during the `yarn build` phase.

File: package.json (hunk 1 above)
Comment (esirK) [reply package.json:127]: ooh yeah. I'll update that

**Pull request merged**